lib/types/local-date.js

```diff
@@ -0,0 +1,200 @@
+"use strict";
+var utils = require('../utils');
+/** @module types */
+
+/** @const */
+var millisecondsPerDay = 86400000;
+/** @const */
+var dateCenter = Math.pow(2,31);
+/**
+ *
+ * Creates a new instance of LocalDate.
+ * @class
+ * @classdesc A date without a time-zone in the ISO-8601 calendar system, such as 2010-08-05.
+ * <p>
+ *   LocalDate is an immutable object that represents a date, often viewed as year-month-day. For example, the value "1st October 2014" can be stored in a LocalDate.
+ * </p>
+ * <p>
+ *   This class does not store or represent a time or time-zone. Instead, it is a description of the date, as used for birthdays. It cannot represent an instant on the time-line without additional information such as an offset or time-zone.
+ * </p>
+ * @param {Number} year The year.
+ * @param {Number} month Between 1 and 12 inclusive.
+ * @param {Number} day Between 1 and the number of days in the given month of the given year.
+ * @constructor
+ */
+function LocalDate(year, month, day) {
```

Comment:
Since LocalDate and LocalTime are currently timezone independent, and the concept of 'local' implies local timezone (at least to me), should we consider renaming these?